dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/ValidationReleasableFactoryImpl.java

```diff
@@ -145,15 +135,199 @@ protected void unsetManagedObjectService(ServiceReference<ManagedObjectService>
+        String parameterNameProviderClassName = bootstrapConfiguration.getParameterNameProviderClassName();
+        ParameterNameProvider pnp = null;
+
+        if (parameterNameProviderClassName == null) {
+            pnp = config.getDefaultParameterNameProvider();
+        } else {
+            @SuppressWarnings("unchecked")
+            Class<? extends ParameterNameProvider> parameterNameProviderClass = (Class<? extends ParameterNameProvider>) loadClass(parameterNameProviderClassName, appClassLoader);
+            pnp = createManagedObject(parameterNameProviderClass);
+        }
+
+        if (pnp != null) {
+            config.parameterNameProvider(pnp);
+        }
+    }
+
+    private void createManagedClockProvider(Configuration<?> config, ClassLoader appClassLoader) {
+        BootstrapConfiguration bootstrapConfiguration = config.getBootstrapConfiguration();
+        String clockProviderClassName = bootstrapConfiguration.getClockProviderClassName();
+        ClockProvider clockProvider = null;
+
+        if (clockProviderClassName == null) {
+            clockProvider = config.getDefaultClockProvider();
+        } else {
+            @SuppressWarnings("unchecked")
+            Class<? extends ClockProvider> clockProviderClass = (Class<? extends ClockProvider>) loadClass(clockProviderClassName, appClassLoader);
+            clockProvider = createManagedObject(clockProviderClass);
+        }
+
+        if (clockProvider != null) {
+            config.clockProvider(clockProvider);
+        }
+    }
+
+    private void createManagedConstraintValidatorFactory(Configuration<?> config, ClassLoader appClassLoader) {
+        BootstrapConfiguration configSource = config.getBootstrapConfiguration();
+        String constraintValidatorFactoryClassName = configSource.getConstraintValidatorFactoryClassName();
+        ConstraintValidatorFactory cvf = null;
+
+        if (constraintValidatorFactoryClassName == null) {
+            // use default
+//            ManagedObject<?> mangedObject = createManagedObject(new ReleasableConstraintValidatorFactory(this));
+//            cvf = (ConstraintValidatorFactory) mangedObject.getObject();
+            cvf = createManagedObject(InjectingConstraintValidatorFactory.class);
+        } else {
+            @SuppressWarnings("unchecked")
+            Class<? extends ConstraintValidatorFactory> constraintValidatorFactoryClass = (Class<? extends ConstraintValidatorFactory>) loadClass(constraintValidatorFactoryClassName,
+                                                                                                                                                  appClassLoader);
+
+            cvf = createManagedObject(constraintValidatorFactoryClass);
+        }
+
+        if (cvf != null) {
+            config.constraintValidatorFactory(cvf);
+        }
+    }
+
+//    private Set<ValueExtractorDescriptor> createValidationXmlValueExtractors(Configuration<?> config, ClassLoader appClassLoader) {
```

Comment:
can this commented out code be removed?

Reply:
No, will be enabling this code in the near future.